doc/contribute/external.rst

```diff
@@ -164,3 +170,49 @@ The flowchart below shows an overview of the process:
    :align: center
 
    Submission process
+
+.. _external-tooling:
+
+Contributing External Tooling
+*****************************
+
+This section deals exclusively with the inclusion of external tooling in the
+Zephyr project, where tooling is defined as software that assists the
+compilation, testing or simulation processes but in no case ends up being part
+of the code compiled and linked into the final image.
+
+Tooling components must be released under a license approved by the
+`Open Source Initiative (OSI)`_.
```

Comment:
Do we want to make any comment here about any exceptions should be raised to the TSC for initial discussion.  This makes it sound like if its not OSI its not ok, which isn't true (as we clearly support commercial tooling at various levels).

Reply:
So we need to either clarify what this section applies to or explain how a commercial compiler applies.   If I was new to the project I would think that this section would say that the project isn't interested or willing to take support for that commercial compiler - since its not OSI licensed approved.